### ISSUE TITLE
Deleting fails for a large map

### ISSUE THREAD

**Issue by jeromedoyle** [jsvisa/elixir_talk#6]
If I have a small map that I encode with Poison, `put`, `reserve`, and `delete`, it works correctly. If I use a large map, it fails deleting.

```
def test_small do
  encoded_notification = Poison.encode!(small_notification)
  ElixirTalk.put(pid, encoded_notification)
  {:reserved, id, {_, content}} = ElixirTalk.reserve(pid, 3600)
  ElixirTalk.delete(pid, id)
  ElixirTalk.quit(pid)
end

def test_large do
  encoded_notification = Poison.encode!(large_notification)
  ElixirTalk.put(pid, encoded_notification)
  {:reserved, id, {_, content}} = ElixirTalk.reserve(pid, 3600)
  ElixirTalk.delete(pid, id)
  ElixirTalk.quit(pid)
end

def small_notification do
  %{small: "notification"}
end

def large_notification do
  notification = %{}
  Enum.map(1..1000, fn(x) ->
    Map.put(notification, :k, "notification_value_#{x}")
  end)
end
```

```
2016-03-03 08:36:03.869 pid=<0.505.0> [error] GenServer #PID<0.505.0> terminating
** (FunctionClauseError) no function clause matching in ElixirTalk.Connect.terminate/2
    (elixir_talk) lib/elixir_talk/connect.ex:83: ElixirTalk.Connect.terminate({:system_limit, [{:erlang, :binary_to_atom, ["ification_value_49\"},{\"k\":\"notification_value_50\"},{\"k\":\"notification_value_51\"},{\"k\":\"notification_value_52\"},{\"k\":\"notification_value_53\"},{\"k\":\"notification_value_54\"},{\"k\":\"notification_value_55\"},{\"k\":\"notification_value_56\"},{\"k\":\"notification_value_57\"},{\"k\":\"notification_value_58\"},{\"k\":\"notification_value_59\"},{\"k\":\"notification_value_60\"},{\"k\":\"notification_value_61\"},{\"k\":\"notification_value_62\"},{\"k\":\"notification_value_63\"},{\"k\":\"notification_value_64\"},{\"k\":\"notification_value_65\"},{\"k\":\"notification_value_66\"},{\"k\":\"notification_value_67\"},{\"k\":\"notification_value_68\"},{\"k\":\"notification_value_69\"},{\"k\":\"notification_value_70\"},{\"k\":\"notification_value_71\"},{\"k\":\"notification_value_72\"},{\"k\":\"notification_value_73\"},{\"k\":\"notification_value_74\"},{\"k\":\"notification_value_75\"},{\"k\":\"notification_value_76\"},{\"k\":\"notification_value_77\"},{\"k\":\"notification_value_78\"},{\"k\":\"notification_value_79\"},{\"k\":\"notification_value_80\"},{\"k\":\"notification_value_81\"},{\"k\":\"notification_value_82\"},{\"k\":\"notification_value_83\"},{\"k\":\"notification_value_84\"},{\"k\":\"notification_value_85\"},{\"k\":\"notification_value_86\"},{\"k\":\"notification_value_87\"},{\"k\":\"notification_value_88\"},{\"k\":\"notification_value_89\"},{\"k\":\"notification_value_90\"},{\"k\":\"notification_value_91\"},{\"k\":\"notification_value_92\"},{\"k\":\"notification_value_93\"},{\"k\":\"notification_value_94\"},{\"k\":\"notification_value_95\"},{\"k\":\"notification_value_96\"},{\"k\":\"notification_value_97", :utf8], []}, {ElixirTalk.Connect, :str_to_atom, 1, [file: 'lib/elixir_talk/connect.ex', line: 133]}, {ElixirTalk.Connect, :handle_call, 3, [file: 'lib/elixir_talk/connect.ex', line: 64]}, {:gen_server, :try_handle_call, 4, [file: 'gen_server.erl', line: 629]}, {:gen_server, :handle_msg, 5, [file: 'gen_server.erl', line: 661]}, {:proc_lib, :init_p_do_apply, 3, [file: 'proc_lib.erl', line: 240]}]}, [socket: #Port<0.10394>])
    (stdlib) gen_server.erl:643: :gen_server.try_terminate/3
    (stdlib) gen_server.erl:809: :gen_server.terminate/7
    (stdlib) proc_lib.erl:240: :proc_lib.init_p_do_apply/3
Last message: {:delete, 16}
State: [socket: #Port<0.10394>]
```

**Comment by jeromedoyle** [comment]
It looks to be failing on `str_to_atom(result)` on line 109 of connect.ex.

For the small notification, result is 
`DELETED` 

On the large notification, result is
`ification_value_49"},{"k":"notification_value_50"},{"k":"notification_value_51"},{"k":"notification_value_52"},{"k":"notification_value_53"},{"k":"notification_value_54"},{"k":"notification_value_55"},{"k":"notification_value_56"},{"k":"notification_value_57"},{"k":"notification_value_58"},{"k":"notification_value_59"},{"k":"notification_value_60"},{"k":"notification_value_61"},{"k":"notification_value_62"},{"k":"notification_value_63"},{"k":"notification_value_64"},{"k":"notification_value_65"},{"k":"notification_value_66"},{"k":"notification_value_67"},{"k":"notification_value_68"},{"k":"notification_value_69"},{"k":"notification_value_70"},{"k":"notification_value_71"},{"k":"notification_value_72"},{"k":"notification_value_73"},{"k":"notification_value_74"},{"k":"notification_value_75"},{"k":"notification_value_76"},{"k":"notification_value_77"},{"k":"notification_value_78"},{"k":"notification_value_79"},{"k":"notification_value_80"},{"k":"notification_value_81"},{"k":"notification_value_82"},{"k":"notification_value_83"},{"k":"notification_value_84"},{"k":"notification_value_85"},{"k":"notification_value_86"},{"k":"notification_value_87"},{"k":"notification_value_88"},{"k":"notification_value_89"},{"k":"notification_value_90"},{"k":"notification_value_91"},{"k":"notification_value_92"},{"k":"notification_value_93"},{"k":"notification_value_94"},{"k":"notification_value_95"},{"k":"notification_value_96"},{"k":"notification_value_97`

**Comment by jsvisa** [comment]
@jeromedoyle Seems BeansTalk Server returns back a large binary `Result`, instead of `DELETED`, am I missing something?

**Comment by jeromedoyle** [comment]
That's what I thought at first, but, if I delete from the terminal using telnet, I get back `DELETED` as expected.

**Comment by jsvisa** [comment]
Which beanstalk release are you using? I will reproduce it later today.

**Comment by jeromedoyle** [comment]
1.10

**Comment by jsvisa** [comment]
I've found what happend, I'll fix it ASAP in the next release.

**Comment by jeromedoyle** [comment]
You rock! What was it?

**Comment by jsvisa** [comment]
Have not recv all available packaged after the server returned. [like here](https://github.com/jsvisa/elixir_talk/blob/master/lib%2Felixir_talk%2Fconnect.ex#L69), so what `deleted` got actually is the packages left by the last `reserve`

**Comment by jeromedoyle** [comment]
That explains why it was a portion of the content I put into the tube. Thanks
for sorting this out.

**Comment by jsvisa** [comment]
Fixed in [1.1.0](https://github.com/jsvisa/elixir_talk/pull/12)
